apps/meteor/client/views/admin/moderation/UserReports/UserReportInfo.tsx

```diff
@@ -0,0 +1,158 @@
+import type { IUser, UserReport } from '@rocket.chat/core-typings';
+import {
+	Box,
+	Callout,
+	StatesAction,
+	StatesActions,
+	StatesIcon,
+	StatesTitle,
+	ContextualbarFooter,
+	FieldGroup,
+	Field,
+	FieldLabel,
+	FieldRow,
+	ContextualbarSkeleton,
+} from '@rocket.chat/fuselage';
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+import { useEndpoint, useToastMessageDispatch, useTranslation } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+import React, { useMemo } from 'react';
+
+import GenericNoResults from '../../../../components/GenericNoResults';
+import UserCard from '../../../../components/UserCard';
+import { useFormatDate } from '../../../../hooks/useFormatDate';
+import ReportReason from '../helpers/ReportReason';
+import UserColumn from '../helpers/UserColumn';
+import UserContextFooter from './UserContextFooter';
+
+const isUser = (obj: any): obj is IUser => {
```

Comment:
???

Reply:
Actually, the data we got from the react query had the `createdAt:string` but the actual types are `createdAt:Date`. To resolve this, I had done an unknown casting initially, `as unknown as IUser`, but changed it back to using a type guard (`isUser`) after the suggestion above, which I feel like better for avoiding unintentional break.

Reply:
You can use `Serialized` type to resolve it. It should be something like: `Serialized<IUser>`
`Serialized` is found on @rocket.chat/core-typings

Reply:
Thank you; this is really helpful.